lib/puppet_x/relay/agent/backend/orchestrator.rb

```diff
@@ -45,7 +45,14 @@ def exec(run, _state_dir, schedule)
               message: e.message,
               body: e.response.body,
             })
-            raise
+            run_results = {
+              error: 'Failed to send request to orchestrator API',
+              message: e.message,
+              response: JSON.parse(e.response.body),
+            }
+            new_state = run.state.to_complete(outcome: 'failure', run_results: run_results)
```

Comment:
I think we should use `outcome: 'error'` here (like we do in `job/exec.rb`) since this isn't actually a failure from running the job itself (user code can distinguish between these).

Reply:
Done.